pkgs/by-name/ar/archipelago/package.nix

```diff
@@ -0,0 +1,43 @@
+{
+  lib,
+  appimageTools,
+  fetchurl,
+  nix-update-script,
+}:
+let
+  pname = "archipelago";
+  version = "0.5.0";
+  src = fetchurl {
+    url = "https://github.com/ArchipelagoMW/Archipelago/releases/download/${version}/Archipelago_${version}_linux-x86_64.AppImage";
+    hash = "sha256-Dw5BBfCthB9xUJXYmnmIwmqlT/L24QZnKyELLjPGNRA=";
+  };
+
+  appimageContents = appimageTools.extractType2 { inherit pname version src; };
+
+in
+appimageTools.wrapType2 {
+  inherit pname version src;
+  extraPkgs = pkgs: [
+    pkgs.xsel
+    pkgs.xclip
+    pkgs.mtdev
```

Comment:
The one issue I have with this is that for all the players that do not play minecraft, this adds an addional JDK to their environment for no reason(and JDKs are large closures, so I would like to avoid them if possible). I would be able to test adding one, and I have no reason as to why it couldn't work, I just don't want to add it for no reason, which is why I only added the core packages to make it function. Maybe it would be better having an "archipelago-minecraft" package that just overlays a JDK on this package? I could add an `extraPackages` variable to this, so a simple `overrideAttrs` call would work.

Reply:
Valid Point. A dedicated variant package or the ability to override would work for me.

Reply:
I've added the ability to override, and an `archipelago-minecraft` package that adds the JDK. Please let me know if this works for you!